pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2316,6 +2316,7 @@ type SSSDSidecarAdditionalFile struct {
 
 // NetworkSpec for Ceph includes backward compatibility code
 // +kubebuilder:validation:XValidation:message="at least one network selector must be specified when using multus",rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))"
+	// +kubebuilder:validation:XValidation:message="the legacy hostNetwork  setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork != true) || (self.provider  ==  '')"
```

Comment:
I **think** this logic makes sense to me, but @subhamkrai is really the VAP MVP. I would suggest that we try to use `== false` instead of `!= true` since that is a condition that is easier to mentally hold when reading code.

Also, you may need to configure your editor to warn you when inputting double spaces. I keep seeing extra spaces where other contributors don't usually have them.

```suggestion
	// +kubebuilder:validation:XValidation:message="the legacy hostNetwork setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork == false) || (self.provider == '')"
```

By my logic, this validates true under the following scenarios, which I think are correct:
- `hostNetwork == false` (and implicitly, when self.provider is any value
- `hostNetwork == true` and `self.provider == ''`

Reply:
This is a line spacing issue. Looking at the CI failure here, it is suggesting a diff to fix the issue.

```
diff pkg/apis/ceph.rook.io/v1/types.go.orig pkg/apis/ceph.rook.io/v1/types.go
--- pkg/apis/ceph.rook.io/v1/types.go.orig
+++ pkg/apis/ceph.rook.io/v1/types.go
@@ -2316,7 +2316,7 @@
 
 // NetworkSpec for Ceph includes backward compatibility code
 // +kubebuilder:validation:XValidation:message="at least one network selector must be specified when using multus",rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))"
-	// +kubebuilder:validation:XValidation:message="the legacy hostNetwork  setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork != true) || (self.provider  ==  '')"
+// +kubebuilder:validation:XValidation:message="the legacy hostNetwork  setting can only be set if the network.provider is set to the empty string",rule="(self.hostNetwork != true) || (self.provider  ==  ”)"
 type NetworkSpec struct {
 	// Provider is what provides network connectivity to the cluster e.g. "host" or "multus".
 	// If the Provider is updated from being empty to "host" on a running cluster, then the operator will automatically fail over all the mons to apply the "host" network settings.
```

The line is indented when it should not be.

Reply:
Thanks, @BlaineEXE  ! I did the corresponding update, but locally I still get this on the test cluster:

```
Events:
  Type     Reason           Age                From                          Message
  ----     ------           ----               ----                          -------
  Warning  ReconcileFailed  4s (x14 over 45s)  rook-ceph-cluster-controller  failed to reconcile CephCluster "rook-ceph/my-cluster". failed to add finalizer: failed to add finalizer "cephcluster.ceph.rook.io" on "my-cluster": CephCluster.ceph.rook.io "my-cluster" is invalid: spec.network: Invalid value: "object": no such key: hostNetwork evaluating rule: the legacy hostNetwork  setting can only be set if the network.provider is set to the empty string
```

Let's see what the CI thinks ...

Reply:
@obnoxxx could you run `make crds` on your branch?

Reply:
@subhamkrai  wrote:
> @obnoxxx could you run `make crds` on your branch?

done. and committed the result+pushed